sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -244,7 +244,11 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       bucketSpec = getBucketSpec,
       options = extraOptions.toMap)
 
-    dataSource.write(mode, df)
+    dataSource.providingClass.newInstance() match {
+      case jdbc: execution.datasources.jdbc.DefaultSource =>
```

Comment:
It looks a new method is introduced. I think we don't necessarily introduce this new function but use the existing interfaces `CreatableRelationProvider` in `interfaces`.